include/caffe/solver.hpp

```diff
@@ -128,6 +128,29 @@ class AdaGradSolver : public SGDSolver<Dtype> {
   DISABLE_COPY_AND_ASSIGN(AdaGradSolver);
 };
 
+
+template <typename Dtype>
+class RMSpropSolver : public SGDSolver<Dtype> {
```

Comment:
The capitalization (everywhere) should be `RMSProp` rather than `RMSprop`.  (I think a global find/replace should do the trick.)